---
layout: article
---

Что же происходит, когда обновление завершается неудачно?

Скажем, у нас первый набор из 3 реплик обновлен успешно, а во втором один из контейнеров выдал ошибку. Если произошёл сбой, дальнейшее обновление приостанавливается, и в результате у нас в кластере будет исполнятся и старая, и новая версии. Так это работает по умолчанию - ставится на паузу и ждёт наших действий. 

Но мы можем изменить поведение, как реагировать на ошибку при обновлении, используя флаг `--update-failure-action`. Его можно выставить в `pause`, `continue` и `rollback`. Из названий понятно, что continue проигнорирует ошибку и продолжит обновлять, а rollback откатит всё на старые версии.

```
docker service update --update-failure-action continue --image=nginx:alpine webserver
```
